connectOpenAIAPI.php

```diff
@@ -1,50 +1,87 @@
+
     include("include/init.php");
     require_once("connectYTAPI.php");
     include("env_constants.php");
 
-    // call function to retrieve video data from YouTube API
-    $youtubeData = getYoutubeData();
+    acceptCorsRequestsFromTheChromeExtension();
 
-    // array that will hold messages to ask ChatGPT
-    $input = [];
+    // save videoId, ThemesToReview, and RangeSlider from GET request as variables
+    $videoId = $_GET['videoId'];
+    $ThemesToReviewJSON = $_GET['reviewThemes'];
+    $ThemesToReview = json_decode($ThemesToReviewJSON, true);
+    $RangeSliderJSON = $_GET['rangeSlider'];
+    $RangeSlider = json_decode($RangeSliderJSON, true);
 
-    foreach($youtubeData as $video){
-        $title = $video[0];
-        $description = $video[1];
-        $categoryId = $video[2];
-        $tags = $video[3];
-        $stringTags = implode(',', $tags);
-        $rating = $video[4];
 
-        array_push($input, "The video has a title of '".$title."' a description of '".$description."' a category id of ".$categoryId." a rating of ".$rating." and the following tags: ".$stringTags);
+    // retrieves video data from YouTube API and returns an array with important content
+    $video = getYoutubeData($videoId);
 
-    }
+    $title = $video['title'];
+    $description = $video['description'];
+    $categoryId = $video['categoryId'];
+    $tags = $video['tags'];
+    $stringTags = implode(',', $tags);
+
+    $rating = $video['rating'];
+
+    // variable that will hold the message to ask ChatGPT
+    $input = "The video has a title of '".$title."' a description of '".$description."' a category id of ".$categoryId." a rating of ".$rating." and the following tags: ".$stringTags;
 
     // send json file, authenticates with key
     $headers = [
         "Content-Type: application/json",
         "Authorization: Bearer ".openAIApiKey
     ];
 
-    // tell OpenAI which model to use, gives message to interpret, temp is response variance
+    //This array maps type codes (what's sent to us from the frontend) with English descriptions for ChatGPT
+    $ContentWarningTypes = [
+        'Profanity' => 'Profanity',
+        'Violence' => 'Violence',
+        'Politics' => 'Political Content',
+        'Horrer' => 'Horrer/Disturbing Themes',
+        'Substances' => 'Substance Use'
+    ];
+
+    $ThemeString = "";
+    foreach($ThemesToReview as $Theme=>$Value) {
+        if($Value){
+            $ThemeString .= "- ".$ContentWarningTypes[$Theme]."\n";
+        }
+    }
+
+
+    $ReturnFormatString = "{\n";
+    foreach($ContentWarningTypes as $Key => $Type){
+        if($ThemesToReview[$Key]){
+             $ReturnFormatString .= "\"$Key\": true/false //".$Type." present or not\n";
+        }
+        else{
+            $ReturnFormatString .= "\"$Key\": null //we aren't checking for this, so ignore it\n";
+        }
+    }
+    $ReturnFormatString .= "\n}";
+
+    
+    
+
+    // send request to OpenAI API with the input as the prompt
```

Comment:
I can't remember if I mentioned this in a previous PR, but I'd make a function called `CallOpenAiApi` or something like that. This file has a lot of boilerplate API code that sort of clutters things up in here. It'd be easier to read this code if the specifics of calling the API were in a separate function.